js-practive/js/main.js

```diff
@@ -1 +1,260 @@
-// Place your code here, or you can split it how you what. It's up to you 😎
\ No newline at end of file
+// console.log(event)
+    // 1.очищення всієї сторінки 
+    // 2.умови, що знаходяться в хещі та їх валідація (асинхронний запит який вибудовує сторінку) (вбудувати сторінку )
+let usersTable
+let createButton
+let userForm 
+
+setUsersTable()
+setHtmlElementUserForm()
+
+function setUsersTable() {
+    usersTable = document.getElementById('users-table-list');
+    createButton = document.getElementById('create');
+
+    createButton.addEventListener('click', goToUserForm);
+    usersTable.addEventListener('click', onUsersTableClick);
+}
+
+function setHtmlElementUserForm() {
+    userForm = document.getElementById('users-edit');
+    userForm.addEventListener('click', onUserFormClick);
+}
+
+readingHashState();
+
+function readingHashState() {
+    addEventListener('popstate', (event) => {
+            
+        switch (location.hash.split("=")[0]) {
+            case '#user-form': {
+                if (window.location.hash.split("=")[1]) {
```

Comment:
you shouldn't use if here. 
I had better write it like one flow and only where neede apply condition.

It will look like 

`
fetch('/user-form.html')
                    .then((res) => res.text())
                    .then((stringHtml) => document.body.innerHTML = stringHtml)
                   .then(() => if if ....  getSingleUser(window.location.hash.split("=")[1]) : )
                    .then((user) => if (user) do code for that user in other case do if there is not user)
               
`